backend/rest/rest.go

```diff
@@ -0,0 +1,215 @@
+
+// Size returns the number of bytes written to the backend so far.
+func (rb *RestBlob) Size() uint {
+	return uint(rb.buf.Len())
+}
+
+// A simple REST backend.
+type Rest struct {
+	url      *url.URL
+	connChan chan struct{}
+}
+
+// Open opens the http backend at the specified url.
+func Open(url *url.URL) (*Rest, error) {
+	connChan := make(chan struct{}, connLimit)
+	for i := 0; i < connLimit; i++ {
+		connChan <- struct{}{}
+	}
+	return &Rest{url: url, connChan: connChan}, nil
+}
+
+// Location returns a string that specifies the location of the repository, like a URL.
+func (b *Rest) Location() string {
+	return b.url.Host
+}
+
+// Create creates a new Blob. The data is available only after Finalize()
+// has been called on the returned Blob.
+func (b *Rest) Create() (backend.Blob, error) {
+	blob := RestBlob{
+		b:   b,
+		buf: &bytes.Buffer{},
+	}
+	return &blob, nil
+}
+
+// Get returns an io.ReadCloser for the Blob with the given name of type t.
+func (b *Rest) Get(t backend.Type, name string) (io.ReadCloser, error) {
+	resp, err := http.Get(restPath(b.url, t, name))
+	if err == nil && resp.StatusCode != 200 {
+		err = errors.New("blob not found")
+	}
+	return resp.Body, err
+}
+
+// GetReader returns an io.ReadCloser for the Blob with the given name of
+// type t at offset and length.
+func (b *Rest) GetReader(t backend.Type, name string, offset, length uint) (io.ReadCloser, error) {
+	req, err := http.NewRequest("GET", restPath(b.url, t, name), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Range", "bytes="+string(offset)+"-"+string(offset+length))
+
+	resp, errg := http.DefaultClient.Do(req)
+	if errg != nil {
+		return nil, errg
+	}
+
+	return backend.LimitReadCloser(resp.Body, int64(length)), nil
+}
+
+// Test a boolean value whether a Blob with the name and type exists.
+func (b *Rest) Test(t backend.Type, name string) (bool, error) {
+	found := false
+
+	req, err := http.NewRequest("HEAD", restPath(b.url, t, name), nil)
+	if err != nil {
+		return found, err
+	}
+
+	resp, errh := http.DefaultClient.Do(req)
+	if resp.StatusCode == 200 {
+		found = true
+	}
+	return found, errh
+}
+
+// Remove removes a Blob with type t and name.
+func (b *Rest) Remove(t backend.Type, name string) error {
+	req, err := http.NewRequest("DELETE", restPath(b.url, t, name), nil)
+	if err != nil {
+		return err
+	}
+
+	_, errd := http.DefaultClient.Do(req)
+	return errd
+}
+
+// Close the backend
+func (b *Rest) Close() error {
+	return nil
+}
+
+// List returns a channel that yields all names of blobs of type t in
+// lexicographic order. A goroutine is started for this. If the channel
+// done is closed, sending stops.
+func (b *Rest) List(t backend.Type, done <-chan struct{}) <-chan string {
+	ch := make(chan string)
+
+	resp, err := http.Get(restPath(b.url, t, ""))
+	if err != nil {
+		close(ch)
+		return ch
+	}
+
+	data, errd := ioutil.ReadAll(resp.Body)
+	if errd != nil {
+		close(ch)
+		return ch
+	}
+
+	var list []string
```

Comment:
Oh no, that's no longer necessary. I've just yesterday started to port the REST backend to the most recent version of the code in master, and I'll continue today. Maybe we'll have an up to date version in the evening. If you like, drop by in `#restic` on freenode IRC.